### ISSUE TITLE
Style/InverseMethods: Auto-correct can cause stack overflow due to recursion

### ISSUE THREAD

**Issue by jbodah** [rubocop-hq/rubocop#6421]
I had a codebase I was auto-correcting and noticed this issue. I would expect it to be an issue with other auto-correct code as well. The source code shouldn't really exist in the first place, but I think auto-correct should also never (or rarely) break code. This may be difficult to avoid generically but I would think that tracking the current context of the auto-correct would be possible (i.e. we know we're already in `!=`)

## Expected behavior

Auto-correct should not produce a new recursive call

## Actual behavior

The user function `!=` caused a recursive call after being auto-corrected (see below for details) which resulted in a stack overflow

## Steps to reproduce the problem

```rb
# hello.rb
class MyClass
  def !=(other)
    !(self == other)
  end
end
```

```sh
$ rubocop --auto-corret
```

```rb
# hello.rb
class MyClass
  def !=(other)
    self != other
  end
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.60.0 (using Parser 2.5.1.2, running on ruby 2.3.2 x86_64-darwin17)
```

**Comment by rrosenblum** [comment]
This is an edge case that most cops wouldn't need to worry about. Even with inverse methods it is an edge case because it is unlikely that the correction is going to be the same as the method itself. I don't think it would be too difficult to look at the node parent to make sure we don't register an offense in a case like this. Unfortunately, it seems like something that would have to be implemented inside of each cop rather than having a global check in the corrector.
